src/order/directed.lean

```diff
@@ -189,6 +214,27 @@ lemma is_bot_iff_is_min [is_directed α (≥)] : is_bot a ↔ is_min a :=
 
 lemma is_top_iff_is_max [is_directed α (≤)] : is_top a ↔ is_max a := ⟨is_top.is_max, is_max.is_top⟩
 
+/-- The property that a function preserves `is_lub` on directed sets. -/
+def preserves_lub_on_directed [preorder β] (f : α → β) : Prop :=
+∀ (d : set α) (a : α), d.nonempty → directed_on (≤) d → is_lub d a → is_lub (f '' d) (f a)
```

Comment:
It's not clear to me whether this definition belongs here, or just in the scott topology files that you add later. I'll let another reviewer decide.

Reply:
@YaelDillies suggested moving it https://github.com/leanprover-community/mathlib4/pull/2508#discussion_r1118178422